crates/nu-command/src/system/run_external.rs

```diff
@@ -418,19 +418,36 @@ impl ExternalCommand {
                             #[cfg(unix)]
                             {
                                 use nu_ansi_term::{Color, Style};
+                                use std::ffi::CStr;
                                 use std::os::unix::process::ExitStatusExt;
+
                                 if x.core_dumped() {
+                                    let cause = x
+                                        .signal()
+                                        .and_then(|sig| unsafe {
+                                            // SAFETY: We should be the first to call `char * strsignal(int sig)`
+                                            let sigstr_ptr = libc::strsignal(sig);
+                                            if sigstr_ptr.is_null() {
+                                                return None;
+                                            }
+
+                                            // SAFETY: The pointer points to a valid non-null string
+                                            let sigstr: &'static CStr = CStr::from_ptr(sigstr_ptr);
```

Comment:
Is `'static` correct here? That means "[that the data pointed to by the reference lives for the entire lifetime of the running program.](https://doc.rust-lang.org/rust-by-example/scope/lifetime/static_lifetime.html)" But looking at [the `strsignal()` docs](https://man7.org/linux/man-pages/man3/strsignal.3.html), they say "The string can be used only until the next call to `strsignal()`."

Maybe we should convert the `strsignal()` result to a `String` to avoid these kinds of issues.

Reply:
Yes, it's worth it. I fixed it